Framework/Algorithms/test/ReflectometryBeamStatisticsTest.h

```diff
@@ -0,0 +1,366 @@
+    TS_ASSERT(run.hasProperty("beam_stats.beam_rms_variation"))
+    TS_ASSERT(!run.hasProperty("beam_stats.bent_sample"))
+    TS_ASSERT(!run.hasProperty("beam_stats.first_slit_angular_spread"))
+    TS_ASSERT(!run.hasProperty("beam_stats.incident_angular_spread"))
+    TS_ASSERT(!run.hasProperty("beam_stats.sample_waviness"))
+    TS_ASSERT(!run.hasProperty("beam_stats.second_slit_angular_spread"))
+  }
+  static void
+  checkReflectedStatisticsContainedInSampleLogs(const API::Run &run) {
+    TS_ASSERT(run.hasProperty("beam_stats.beam_rms_variation"))
+    TS_ASSERT(run.hasProperty("beam_stats.bent_sample"))
+    TS_ASSERT(run.hasProperty("beam_stats.first_slit_angular_spread"))
+    TS_ASSERT(run.hasProperty("beam_stats.incident_angular_spread"))
+    TS_ASSERT(run.hasProperty("beam_stats.sample_waviness"))
+    TS_ASSERT(run.hasProperty("beam_stats.second_slit_angular_spread"))
+  }
+
+  void checkWrongSlitsThrows(const int slit) {
+    const std::string slit1 = slit == 1 ? "non-existent" : "slit1";
+    const std::string slit2 = slit == 2 ? "non-existent" : "slit2";
+    auto reflectedWS = makeWS(0.7);
+    const std::vector<int> reflectedForeground{FGD_FIRST, BEAM_CENTRE,
+                                               FGD_LAST};
+    auto directWS = makeWS(0.);
+    const std::vector<int> directForeground{FGD_FIRST, BEAM_CENTRE, FGD_LAST};
+    Algorithms::ReflectometryBeamStatistics alg;
+    alg.setChild(true);
+    alg.setRethrows(true);
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("ReflectedBeamWorkspace", reflectedWS))
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("ReflectedForeground", reflectedForeground))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("DirectBeamWorkspace", directWS))
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("DirectForeground", directForeground))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("PixelSize", PIXEL_SIZE))
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("DetectorResolution", DET_RESOLUTION))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("FirstSlitName", slit1))
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("FirstSlitSizeSampleLog", "slit1.size"))
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("SecondSlitName", slit2))
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("SecondSlitSizeSampleLog", "slit2.size"))
+    TS_ASSERT_THROWS_EQUALS(
+        alg.execute(), Kernel::Exception::NotFoundError & e, e.what(),
+        std::string("Could not find slit search object non-existent"))
+    TS_ASSERT(!alg.isExecuted())
+  }
+
+  static API::MatrixWorkspace_sptr makeWS(const double braggAngle) {
+    using namespace WorkspaceCreationHelper;
+    using Geometry::deg2rad;
+    constexpr double startX{1000.};
+    const Kernel::V3D sourcePos{0., 0., -L1};
+    const Kernel::V3D &monitorPos = sourcePos;
+    const Kernel::V3D samplePos{
+        0.,
+        0.,
+        0.,
```

Comment:
Last comma?

Reply:
It's there for consistency and expandability. What if we ever get a fourth dimension?

Reply:
For me, V3D is a point with three coordinates (x, y, z) and not a vector.

Reply:
Thus, I would call it P3D and for a fourth dimension P4D, following the logic.

Reply:
Seeing V3D as a vector: it's not a general vector